Files/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,255 @@
+﻿using Files.Helpers;
+using Files.UserControls.MultitaskingControl;
+using Files.Views;
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Mvvm.Input;
+using Microsoft.Toolkit.Uwp;
+using System;
+using System.Linq;
+using System.Windows.Input;
+using Windows.System;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Navigation;
+using static Files.Views.MainPage;
+
+namespace Files.ViewModels
+{
+    public class MainPageViewModel : ObservableObject
+    {
+        #region Private Members
```

Comment:
@d2dyno1 FYI: I'd prefer we don't use regions in the code.

Reply:
Why? 😧
They really help a lot with grouping code, making it look clean. Moreover, you can collapse a #region section in VS leaving the focus on the code that's important to you.

Reply:
@yaichenbaum It doesn't affect the performance. It only makes code look more clean and doesn't make it look like one big mess which is very apparent with `ItemViewModel` and `Interaction` classes.

Reply:
>  we shouldn't be using regions.

Why is that? @yaichenbaum

Reply:
:shipit:

Reply:
1. Regions are painful for other developers inspecting your the first time. They first have to expand all the collapsed sections to get the full picture of the code. And because of all the #regionand #endregion lines the code gets even larger.
2. Regions often lead to bad code because developers easily put a region around larger code blocks to make them hide instead of thinking how to structure their code well.
[Source](https://marcduerst.com/2016/10/03/c-regions-are-evil/)

Reply:
These are not viable points for me. Addressing the first point, regions can be easily expanded and collapsed by following shortcuts: `Ctrl + M, P` and `Ctrl + M, O`.
> And because of all the #regionand #endregion lines the code gets even larger.

Assuming you'll have an average maximum of 10 region directives, it doesn't make the code larger.

re 2.
It really all depends on the person who wrote the code. For instance the `ItemViewModel` class has some massive functions of which the complexity is "Very High" (CodeFactor) and there aren't any `#region`s enclosements used. In other words, the code doesn't degrade with `region` directives use.

That being said, I'll remove #regions from this PR, but the arguments brought by me **should** be taken into consideration. @tsvietOK @yaichenbaum

Reply:
(@tsvietOK the opposite of article referenced by you say [C# Corner]("Using C# Regions To Improve Code Readability" https://www.c-sharpcorner.com/article/using-regions-to-improve-code-readability/))

Reply:
I'll remove regions in this PR because I don't think my opinion matters in the end, because a "quick" update to the codestyle guidelines will probably soon roll-out as it did before from my past experience

Reply:
@d2dyno1 Your opinion matters. We make changes to code style guidelines based on the consensus of contributors and to prevent mistakes like this from happening after PRs have been opened initially.

Regarding regions, I think there are better ways we can simplify the code besides hiding entire blocks completely. Clean code should be more than aesthetics and you'll find that most other languages don't rely on the concept of regions. With that said, I think we should work together to compromise on a new code style guideline for regions that maybe allows them for very specific situations, but not everywhere in every file. Perhaps we can think of them as a way to group _some_ functionality instead of assign names to already self-explanatory constructs.